examples/http/onvif_camera.js

```diff
@@ -0,0 +1,324 @@
+ * @returns {string} The Digest Authorization header.
+ */
+function generateDigestAuth(realm, nonce, uri, method) {
+    var hash1 = D.crypto.hash(username + ":" + realm + ":" + password, "md5", "utf8", "hex");
+    var hash2 = D.crypto.hash(method + ":" + uri, "md5", "utf8", "hex");
+    var response = D.crypto.hash(hash1 + ":" + nonce + ":" + hash2, "md5", "utf8", "hex");
+    var authHeader = 'Digest username="' + username + '", realm="' + realm + '", nonce="' + nonce + '", uri="' + uri + '", response="' + response + '"';
+    return authHeader;
+}
+
+/**
+ * Checks the health of the camera connection.
+ * @param {Array} streamUrisProfiles Array of objects containing profile names and corresponding stream URIs.
+ * @returns A promise that resolves with an array of health statuses.
+ */
+function healthCheck(streamUrisProfiles) {
+    var healthPromises = [];
+
+    streamUrisProfiles.forEach(function(stream) {
+        var profileName = stream.profileName;
+        var uris = stream.uris;
+        uris.forEach(function(uri) {
+            var d = D.q.defer();
+            getNonceAndRealm(uri)
+                .then(function(digestParams) {
+                    var authHeader = generateDigestAuth(digestParams.realm, digestParams.nonce, uri, "DESCRIBE");
+                    var command = "DESCRIBE " + uri + " RTSP/1.0\r\nCSeq: 2\r\nAuthorization: " + authHeader + "\r\n";
+                    telnetParams.command = command;
+                    telnetParams.timeout = 1000;
+                    D.device.sendTelnetCommand(telnetParams, function(out, err) {
+                        if (err) {
+                            console.log(err);
+                        } else {
+                            d.resolve({ healthStatus: out, profileName: profileName });
+                        }
+                    });
+                }).catch(function(error) {
+                    console.log(error);
+                    d.resolve({ healthStatus: "NOT OK", profileName: profileName });
+                });
+
+            healthPromises.push(d.promise);
+        });
+    });
+    return D.q.all(healthPromises);
+}
+
+function sanitize(output){
+    var recordIdReservedWords = ['\\?', '\\*', '\\%', 'table', 'column', 'history'];
+    var recordIdSanitisationRegex = new RegExp(recordIdReservedWords.join('|'), 'g');
+    return output.replace(recordIdSanitisationRegex, '').slice(0, 50).replace(/\s+/g, '-').toLowerCase();
+}
+
+// Retrieves the capturing status and populates the Custom Driver Table
+function getCapturingStatus(data){
+    data.forEach(function(response, index) {
+        var status;
+        if (response.healthStatus.indexOf("200 OK") != -1) {
+            status = "OK";
+        } else {
+            status = "NOT OK"; 
+        } 
+         
+        var name = response.profileName; 
+        var recordId = (index + 1) + "-" + name;
+
+        if (profileName[0].toLowerCase() === "all" || profileName.indexOf(name) !== -1 ) {
```

Comment:
The filtering MUST happen on the data collection phase not the data visualisation.

The way its currently implemented we will always get ALL the profiles and just filter at the end and for cameras with large amount of profiles the script will be extremely slow and may never complete even if the customer filters on a single profile